### PR TITLE
Room css fixes

### PR DESCRIPTION
**Pull request by SethDonohue**
Added CSS Grids for room to control buttons and active voter info>
 - Contains media query for grid span changes and sizing
- Works on mobile, and responsive

Merged in savedButton conditional logic, saveRoom css and TODO's
removed unnecessary imports from a few components

### CODE REVIEW

File: src/component/app/_app.scss [hunk 1]
```diff
@@ -1,5 +1,45 @@
...
+  position: absolute;
+  margin: 0.5vw;
+}
+
+.login-logout-button {
+  top: 0;
+  right: 0;
+  width: 40%;
+  @media only screen and (min-width: 600px) {
+    right: 0;
```
Comment (RobertMcReed) [inline src/component/app/_app.scss]: Doesn't look like this `right: 0` is needed, since it has that before the media query
@SethDonohue

File: src/component/app/_app.scss [hunk 2]
```diff
@@ -1,5 +1,45 @@
...
+  left: 0;
+  width: 40%;
+  .saved-button {
+    width: 100%;
+  }
+
+  @media only screen and (min-width: 600px) {
+    width: 20%;
+    left: 55%;
+    .saved-button {
```
Comment (RobertMcReed) [inline src/component/app/_app.scss]: Ditto here, that .saved-button doesn't appear to be needed

File: src/component/material-ui/nav-wrapper/index.js [hunk 3]
```diff
@@ -12,12 +14,17 @@ class NavWrapper extends Component {
       anchorEl: null,
       dialogOpen: true,
       popoverOpen: false,
+      viewingSaved: false,
     }
 
   handleToggle = () => {
```
Comment (RobertMcReed) [inline src/component/material-ui/nav-wrapper/index.js]: My understanding is that this should be setup with a callback:

```
this.setState(prevState => ({
  open: !prevState.open,
});
```

File: src/component/material-ui/nav-wrapper/index.js [hunk 4]
```diff
@@ -12,12 +14,17 @@ class NavWrapper extends Component {
...
     this.setState({ open: !this.state.open });
   }
 
+  toggleViewingSaved = () => {
+    this.setState({ viewingSaved: !this.state.viewingSaved });
```
Comment (RobertMcReed) [inline src/component/material-ui/nav-wrapper/index.js]: Same with this one:

```
this.setState(prevState => ({
  viewingSaved: !prevState.viewingSaved,
});
```

File: src/component/material-ui/nav-wrapper/index.js (hunk 3 above)
Comment (RobertMcReed) [inline src/component/material-ui/nav-wrapper/index.js]: we should probably make this more specific since there are multiple toggles now.

File: src/component/material-ui/nav-wrapper/index.js [hunk 5]
```diff
@@ -40,6 +47,39 @@ class NavWrapper extends Component {
     this.setState({ popoverOpen: !this.state.popoverOpen });
   }
 
+  renderSavedSwitch = (location) => {
+    switch (location) {
+      case '/saved':
+        return (
+          <Link
+            to="/" 
+            href="/"
```
Comment (RobertMcReed) [inline src/component/material-ui/nav-wrapper/index.js]: Why do we have the `href` here. With `Link` isn't it just supposed to have the `to`?

File: src/component/material-ui/nav-wrapper/index.js [hunk 6]
```diff
@@ -87,25 +128,25 @@ class NavWrapper extends Component {
...
     const loginSignupJSX = (
       <RaisedButton
-        label="Signup/Login"
+        className="login-logout-button"      
         onClick={(event) => this.setState({
           anchorEl: event.target,
           popoverOpen: true,
         })}
-      />
+      >
```
Comment (RobertMcReed) [inline src/component/material-ui/nav-wrapper/index.js]: Why did you change this? My understanding was that with Material UI you weren't supposed to put the names inside of the element, you were supposed to use the `label` instead. This should be switched back. (At least that's what Josh said [I think])

File: src/component/material-ui/nav-wrapper/index.js [hunk 7]
```diff
@@ -87,25 +128,25 @@ class NavWrapper extends Component {
...
 
     const authButton = this.props.loggedIn ? logoutJSX : loginSignupJSX;
-
+    
+    // Seth - Using location.pathname to conditional render savedButton with switch function
+    const savedButton = this.props.loggedIn ?
+      this.renderSavedSwitch(this.props.location.pathname) : null;
+    
     return (
-      <Paper
```
Comment (RobertMcReed) [inline src/component/material-ui/nav-wrapper/index.js]: Why are you removing the paper? Isn't that what gives it the theming to keep it consistent with our current setup?

File: src/component/material-ui/nav-wrapper/index.js [hunk 8]
```diff
@@ -173,5 +214,5 @@ const mapDispatchToProps = dispatch => ({
   logout: () => dispatch(logoutAction()),
 });
 
-export default connect(mapStateToProps, mapDispatchToProps)(NavWrapper);
+export default withRouter(connect(mapStateToProps, mapDispatchToProps)(NavWrapper));
```
Comment (RobertMcReed) [inline src/component/material-ui/nav-wrapper/index.js]: withRouter gives it access to `this.props.location`?

File: src/component/room/_room.scss [hunk 9]
```diff
@@ -1,15 +1,60 @@
...
+// }
+
+.info-button-container {
+  display: grid;
+  grid-template-columns: repeat(4, 25%);
+  grid-template-rows: repeat(4, 25%);
+}
+
+.active-voters {
+  grid-row: 1 / span 1;
```
Comment (RobertMcReed) [inline src/component/room/_room.scss]: what is `span`? How does that work, is that supposed to be a var?

File: src/component/room/_room.scss [hunk 10]
```diff
@@ -1,15 +1,60 @@
...
+  margin: $default-button-margin;
+  font-size: 1.15em;
+  font-family: $font-main;
+  @media only screen and (min-width: 600px) {
+    grid-column: 1 / span 1;
+  }
+}
+
+.close-save-button {
+  align-self: center;
```
Comment (RobertMcReed) [inline src/component/room/_room.scss]: What is this? Haven't seen it before. Teach me!

File: src/component/room/index.js [hunk 11]
```diff
@@ -1,7 +1,7 @@
 import { connect } from 'react-redux';
 import React, { Component, Fragment } from 'react';
-import { BrowserRouter, Route, Link } from 'react-router-dom';
 import { RaisedButton, FlatButton, Dialog, Paper } from 'material-ui';
+import { BrowserRouter, Route, Link } from 'react-router-dom';
```
Comment (RobertMcReed) [inline src/component/room/index.js]: Why unschwoop?

File: src/component/material-ui/nav-wrapper/index.js (hunk 3 above)
Comment (SethDonohue) [reply src/component/material-ui/nav-wrapper/index.js]: This viewingSaved state is not longer needed, it was what i forgot to remove. I am using the location.pathname as the check for this logic now.

so the  
```
toggleViewingSaved = () => {
    this.setState({ viewingSaved: !this.state.viewingSaved });
  }
```

can be removed as well

File: src/component/material-ui/nav-wrapper/index.js (hunk 4 above)
Comment (SethDonohue) [reply src/component/material-ui/nav-wrapper/index.js]: This viewingSaved state is not longer needed, it was what i forgot to remove. I am using the location.pathname as the check for this logic now.

so the  
```
toggleViewingSaved = () => {
    this.setState({ viewingSaved: !this.state.viewingSaved });
  }
```

can be removed as well

File: src/component/material-ui/nav-wrapper/index.js (hunk 3 above)
Comment (SethDonohue) [reply src/component/material-ui/nav-wrapper/index.js]: removed, not needed, forgot to pull it out, same as above

File: src/component/material-ui/nav-wrapper/index.js (hunk 7 above)
Comment (SethDonohue) [reply src/component/material-ui/nav-wrapper/index.js]: the paper was put in for the nave bar, but then the background and shadow were set to transparent so it was acting just live a normal div.

Comment (SethDonohue) [reply src/component/material-ui/nav-wrapper/index.js]: The Paper is essentially just used to get that "paper" look

File: src/component/material-ui/nav-wrapper/index.js (hunk 8 above)
Comment (SethDonohue) [reply src/component/material-ui/nav-wrapper/index.js]: yes, it does this for the nav-wrapper so it has access since it is not in a Route

File: src/component/room/_room.scss (hunk 9 above)
Comment (SethDonohue) [reply src/component/room/_room.scss]: span tells it how many rows to go to after starting at row 1. There are many ways to do it, especially when it only goes one gridline, but i wanted to keep them a consistent format of " gridlinestartt# / span gridlinestospan#"

File: src/component/room/_room.scss (hunk 10 above)
Comment (SethDonohue) [reply src/component/room/_room.scss]: haha can't in a comment!

File: src/component/room/index.js (hunk 11 above)
Comment (SethDonohue) [reply src/component/room/index.js]: I saw that... i already fixed that before this comment hahah

File: src/component/app/_app.scss (hunk 1 above)
Comment (SethDonohue) [reply src/component/app/_app.scss]: These must have been left over from me playing around in css yesterday eve. It took a bit to get the dang buttons where they are.

File: src/component/app/_app.scss (hunk 2 above)
Comment (SethDonohue) [reply src/component/app/_app.scss]: same as above, will be removed in next commits

File: src/component/material-ui/nav-wrapper/index.js (hunk 5 above)
Comment (SethDonohue) [reply src/component/material-ui/nav-wrapper/index.js]: for some reason out eslint required me to put it in, otherwise it would have a red error in our file, which is annoying.
We could change the eslint rule, or just put it in.

File: src/component/material-ui/nav-wrapper/index.js (hunk 6 above)
Comment (SethDonohue) [reply src/component/material-ui/nav-wrapper/index.js]: Well we did it this way in nearly everything else, and didn't use the label or primaryText prop, so i made it consistent this way with what we've done everywhere else.

we can go back through and fix this with another cleanup/branch, i'll add a TODO for that.
